### ISSUE TITLE
chef-solo on Windows 7 guests fails with a "No such file or directory" error

### ISSUE THREAD

**Issue by npiguet** [mitchellh/vagrant#5821]
I'm trying to provision a window 7 guest using a vagrant-chef s from an ubuntu 14.04 host. After lots of trial an error I was finally successful in getting vagrant to connect to the windows box with winrm and start chef. 

The chef run appears to start normally, but fails after a few seconds with the following error: 
```
FATAL: Errno::ENOENT: No such file or directory - /tmp/vagrant-chef/1fb3b2abe57ad32207a33de3e507c5d0/cookbooks
```

I went to look for that particular folder on the guest's drive using powershell, and here is what I get:

```
PS C:\tmp\vagrant-chef\1fb3b2abe57ad32207a33de3e507c5d0> cmd /c dir C:\tmp\vagrant-chef\1fb3b2abe57ad32207a33de3e507c5d0
/AD
 Volume in drive C has no label.
 Volume Serial Number is E0CE-337D

 Directory of C:\tmp\vagrant-chef\1fb3b2abe57ad32207a33de3e507c5d0

06/11/2015  09:21 AM    <DIR>          .
06/11/2015  09:21 AM    <DIR>          ..
06/11/2015  09:21 AM    <SYMLINKD>     cookbooks [\\vboxsrv\v-csc-1]
               0 File(s)              0 bytes
               3 Dir(s)  124,502,810,624 bytes free

PS C:\tmp\vagrant-chef\1fb3b2abe57ad32207a33de3e507c5d0> Get-ChildItem -Path .\cookbooks
Get-ChildItem : The network name cannot be found.
At line:1 char:14
+ Get-ChildItem <<<<  -Path .\cookbooks
    + CategoryInfo          : ReadError: (C:\tmp\vagrant-...7c5d0\cookbooks:String) [Get-ChildItem], IOException
    + FullyQualifiedErrorId : DirIOError,Microsoft.PowerShell.Commands.GetChildItemCommand

PS C:\tmp\vagrant-chef\1fb3b2abe57ad32207a33de3e507c5d0>
```

So that directory is a symlink (junction point?) to `\\vboxsrv\v-csc-1`, but this particular share does not seem to exist. 

Note that vagrant does manage to create and mount the share `\\vboxsrv\vagrant\` to `c:\vagrant`. But for some reason this other one is missing.

Also of note: Currently, on my host system my various cookbooks do not reside in the same folder than VagrantFile and are not in a subfolder of that folder either.

Any idea of what's going wrong here

```
Full Configuration: 
Host: 
 - ubuntu 14.04
 - vagrant 1.7.2
 - virtualbox 4.2.24
Guest:
 - Windows 7 (modified from microsoft's own modern.ie vagrant boxes)
 - chef 11.18.12
```

**Comment by npiguet** [comment]
After further investigation, it seems that the issue from symlinks on the host machine:

if I use:
```
chef.cookbooks_path = '/absolute_path_no_symlinks/cookbooks'
chef.roles_path     = '/absolute_path_no_symlinks/roles'
```
or 
```
chef.cookbooks_path = './relative_path_no_symlinks/cookbooks'
chef.roles_path     = './relative_path_no_symlinks/roles'
```
then vagrant's logs show:
```
==> seleniumie8: Mounting shared folders...
    seleniumie8: /vagrant => /vagrant_folder
    seleniumie8: /tmp/vagrant-chef/ecd41b1fcf160c0b58e949ec16757615/roles => /absolute_path_no_symlinks/roles
    seleniumie8: /tmp/vagrant-chef/1fb3b2abe57ad32207a33de3e507c5d0/cookbooks => /absolute_path_no_symlinks/cookbooks
==> seleniumie8: Running provisioner: chef_solo...
```

but if there is any symlink in the cookbooks_path, then vagrant silently fails to mount the `cookbooks` and `roles` shared folder.
```
        chef.cookbooks_path = '/symlink/path/cookbooks'
        chef.roles_path = '/symlink/path/roles'
```
yields
```
==> seleniumie8: Mounting shared folders...
    seleniumie8: /vagrant => /home/dev-admin/workspace/fox/platform/integration-tests/src/main/deploy/profiles/seleniumgrid
==> seleniumie8: Running provisioner: chef_solo...
```
As you can see, the shared folders for cookbooks and roles are not created.

It seems that this case should not fail to provision my VM. And even if it does, it clearly should fail early with a clearer error message

**Comment by sethvargo** [comment]
Fixed by #5913!
